### PR TITLE
Using persistent volumes, fixing build time.

### PR DESCRIPTION
**Pull request by ivannaranjo**
This PR uses persistent volumes to store the packages for the apps in the samples, a much better way to do the build, without ever colliding with the app's sources.

This PR also fixes the build time, ensuring that the SDK does not try to prime the cache multiple times, by setting the `DOTNET_SKIP_FIRST_TIME_EXPERIENCE` environment variable.

### CODE REVIEW

File: csharp/dotnet/Dockerfile [hunk 1]
```diff
@@ -34,4 +35,9 @@ RUN cd /configs/config-1.0.0/ && dotnet new && \
     cd /configs/config-2.0.0/ && dotnet new && \
     rm -rf /configs
 
+# Ensure that no more cache expansion is performed from now on, see
```
Comment (bendory) [inline csharp/dotnet/Dockerfile]: Grammar nit: "Ensure that no **additional** expansion is performed..."

File: csharp/dotnet/examples/TestApp-2.0/cloudbuild.yaml [hunk 2]
```diff
@@ -1,13 +1,18 @@
 # Example build script to create a Docker image from a .NET Core 2.0 Web Api app.
 steps:
 
-# Restore and publish the app. The restore must use a local directory
-# inside of the workspace to preserve the packages across invokations
-# of the dotnet tool.
+# Use a persistent volume to store the restored packages for this app. The
+# packages will available in between calls to the 'dotnet' builder
```
Comment (bendory) [inline csharp/dotnet/examples/TestApp-2.0/cloudbuild.yaml]: wording / clarity: "...will **be** available **across** calls..."

File: csharp/dotnet/examples/TestApp-2.0/cloudbuild.yaml [hunk 3]
```diff
@@ -1,13 +1,18 @@
 # Example build script to create a Docker image from a .NET Core 2.0 Web Api app.
 steps:
 
-# Restore and publish the app. The restore must use a local directory
-# inside of the workspace to preserve the packages across invokations
-# of the dotnet tool.
+# Use a persistent volume to store the restored packages for this app. The
```
Comment (bendory) [inline csharp/dotnet/examples/TestApp-2.0/cloudbuild.yaml]: word choice nit: perhaps "...to **host** the restored packages..." -- "store the restored" sounds redundant to me

**Comment by ivannaranjo** [comment]
Addressed the comments, PTAL.

**Comment by ivannaranjo** [comment]
Gentle ping.

**Comment by ivannaranjo** [comment]
Ping :)

**Pull request merged**